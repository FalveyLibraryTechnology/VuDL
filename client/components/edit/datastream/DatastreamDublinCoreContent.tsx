import React, { useEffect } from "react";
import Button from "@mui/material/Button";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import Grid from "@mui/material/Grid";
import { useEditorContext } from "../../../context/EditorContext";
import { useDublinCoreMetadataContext } from "../../../context/DublinCoreMetadataContext";
import useDatastreamOperation from "../../../hooks/useDatastreamOperation";
import DatastreamDublinCoreValues from "./DatastreamDublinCoreValues";
import DatastreamDublinCoreAddButtons from "./DatastreamDublinCoreAddButtons";
import ObjectPreviewButton from "../ObjectPreviewButton";

const DatastreamDublinCoreContent = (): React.ReactElement => {
    const {
        state: { currentPid, objectDetailsStorage },
        action: { toggleDatastreamModal },
    } = useEditorContext();
    const {
        state: { currentDublinCore },
        action: { setCurrentDublinCore },
    } = useDublinCoreMetadataContext();
    const { uploadDublinCore } = useDatastreamOperation();
    const loaded = Object.prototype.hasOwnProperty.call(objectDetailsStorage, currentPid);
    useEffect(() => {
        if (loaded) {
            // Make a deep clone of the metadata; we don't want to modify anything by reference!
            setCurrentDublinCore(JSON.parse(JSON.stringify(objectDetailsStorage[currentPid].metadata ?? {})));
        }
    }, [loaded]);

    return (
        <>
            <DialogContent>
                <Grid container spacing={1}>
                    <Grid item xs={8}>
                        <DatastreamDublinCoreValues />
                    </Grid>
                    <Grid item xs={4}>
                        <DatastreamDublinCoreAddButtons />
                        <h3>Other Tools</h3>
                        <ObjectPreviewButton pid={currentPid} />
                    </Grid>
                </Grid>
            </DialogContent>

            <DialogActions>
                <Button
                    className="uploadDublinCoreButton"
                    onClick={async () => {
                        await uploadDublinCore(currentDublinCore);
                    }}
                >
                    Save
                </Button>
                <Button onClick={toggleDatastreamModal}>Cancel</Button>
            </DialogActions>
        </>
    );
};

export default DatastreamDublinCoreContent;
