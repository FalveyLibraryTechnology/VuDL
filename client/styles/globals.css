:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
