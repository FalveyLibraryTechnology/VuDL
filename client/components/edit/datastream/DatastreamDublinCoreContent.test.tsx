import React from "react";
import { describe, afterEach, expect, it, jest } from "@jest/globals";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import DatastreamDublinCoreContent from "./DatastreamDublinCoreContent";
import Button from "@mui/material/Button";
import { act } from "react-dom/test-utils";

const mockUseEditorContext = jest.fn();
jest.mock("../../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));

const mockUseDublinCoreMetadataContext = jest.fn();
jest.mock("../../../context/DublinCoreMetadataContext", () => ({
    useDublinCoreMetadataContext: () => {
        return mockUseDublinCoreMetadataContext();
    },
}));

const mockUseDatastreamOperation = jest.fn();
jest.mock("../../../hooks/useDatastreamOperation", () => () => mockUseDatastreamOperation());

jest.mock("./DatastreamDublinCoreValues", () => () => "DatastreamDublinCoreValues");
jest.mock("./DatastreamDublinCoreAddButtons", () => () => "DatastreamDublinCoreAddButtons");

describe("DatastreamDublinCoreContent ", () => {
    let dcValues;
    let editorValues;
    let pid;
    let uploadDublinCore;

    beforeEach(() => {
        pid = "foo";
        editorValues = {
            state: {
                currentPid: pid,
                objectDetailsStorage: {},
            },
            action: {
                toggleDatastreamsModel: jest.fn(),
            },
        };
        mockUseEditorContext.mockReturnValue(editorValues);
        uploadDublinCore = jest.fn();
        mockUseDatastreamOperation.mockReturnValue({ uploadDublinCore });
        dcValues = {
            state: {
                currentDublinCore: {},
            },
            action: {
                setCurrentDublinCore: jest.fn(),
            },
        };
        mockUseDublinCoreMetadataContext.mockReturnValue(dcValues);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("renders", () => {
        const wrapper = shallow(<DatastreamDublinCoreContent />);

        expect(toJson(wrapper)).toMatchSnapshot();
        // Our default data has nothing loaded, and current DC shouldn't be set
        // until data loads.
        expect(dcValues.action.setCurrentDublinCore).not.toHaveBeenCalled();
    });

    it("correctly loads data", () => {
        const metadata = { "dc:title": ["foo"] };
        editorValues.state.objectDetailsStorage[pid] = { metadata };
        mount(<DatastreamDublinCoreContent />);
        expect(dcValues.action.setCurrentDublinCore).toHaveBeenCalledWith(metadata);
    });

    it("correctly defaults to empty data when none is provided", () => {
        editorValues.state.objectDetailsStorage[pid] = {};
        mount(<DatastreamDublinCoreContent />);
        expect(dcValues.action.setCurrentDublinCore).toHaveBeenCalledWith({});
    });

    it("correctly uploads data", () => {
        const metadata = { "dc:title": ["foo"] };
        dcValues.state.currentDublinCore = metadata;
        const wrapper = mount(<DatastreamDublinCoreContent />);
        act(() => {
            wrapper.find(Button).at(0).props().onClick();
        });
        expect(uploadDublinCore).toHaveBeenCalledWith(metadata);
    });
});
