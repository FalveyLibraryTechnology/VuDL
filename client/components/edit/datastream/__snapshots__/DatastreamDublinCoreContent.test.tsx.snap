// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatastreamDublinCoreContent  renders 1`] = `
[
  <div
    className="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
      >
        DatastreamDublinCoreValues
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        DatastreamDublinCoreAddButtons
        <h3>
          Other Tools
        </h3>
        ObjectPreviewButton
      </div>
    </div>
  </div>,
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing css-knqc4i-MuiDialogActions-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium uploadDublinCoreButton css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Save
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Cancel
    </button>
  </div>,
]
`;
